vxquery-core/src/main/java/org/apache/vxquery/metadata/VXQueryCollectionOperatorDescriptor.java

```diff
@@ -82,31 +107,138 @@ public IOperatorNodePushable createPushRuntime(IHyracksTaskContext ctx,
                 fta.reset(buffer);
                 String collectionModifiedName = collectionName.replace("${nodeId}", nodeId);
-                File collectionDirectory = new File(collectionModifiedName);
-
-                // Go through each tuple.
-                if (collectionDirectory.isDirectory()) {
-                    for (int tupleIndex = 0; tupleIndex < fta.getTupleCount(); ++tupleIndex) {
-                        @SuppressWarnings("unchecked")
-                        Iterator<File> it = FileUtils.iterateFiles(collectionDirectory, new VXQueryIOFileFilter(),
-                                TrueFileFilter.INSTANCE);
-                        while (it.hasNext()) {
-                            File xmlDocument = it.next();
-                            if (LOGGER.isLoggable(Level.FINE)) {
-                                LOGGER.fine("Starting to read XML document: " + xmlDocument.getAbsolutePath());
+                if (!collectionModifiedName.contains("hdfs:/")) {
+                    File collectionDirectory = new File(collectionModifiedName);
+                    //check if directory is in the local file system
+                    if (collectionDirectory.exists()) {
+                        System.out.println("local");
+                        // Go through each tuple.
+                        if (collectionDirectory.isDirectory()) {
+                            for (int tupleIndex = 0; tupleIndex < fta.getTupleCount(); ++tupleIndex) {
+                                Iterator<File> it = FileUtils.iterateFiles(collectionDirectory,
+                                        new VXQueryIOFileFilter(), TrueFileFilter.INSTANCE);
+                                while (it.hasNext()) {
+                                    File xmlDocument = it.next();
+                                    if (LOGGER.isLoggable(Level.FINE)) {
+                                        LOGGER.fine("Starting to read XML document: " + xmlDocument.getAbsolutePath());
+                                    }
+                                    parser.parseElements(xmlDocument, writer, fta, tupleIndex);
+                                }
                             }
-                            parser.parseElements(xmlDocument, writer, fta, tupleIndex);
+                        } else {
+                            throw new HyracksDataException("Invalid directory parameter (" + nodeId + ":"
+                                    + collectionDirectory.getAbsolutePath() + ") passed to collection.");
                         }
                     }
                 } else {
-                    throw new HyracksDataException("Invalid directory parameter (" + nodeId + ":"
-                            + collectionDirectory.getAbsolutePath() + ") passed to collection.");
+                    collectionModifiedName = collectionModifiedName.replaceAll("hdfs:/", "");
+                    // Else check in HDFS file system
```

Comment:
Is there a setting required to be set in our config file for this code path? if so add a check.